upstream_utils/upstream_utils.py

```diff
@@ -427,6 +395,48 @@ def replace_tag(self, tag):
         with open(path, "w") as file:
             file.writelines(lines)
 
+    def replace_patch_list(self, patch_list):
+        """Replaces the patch list in the script.
+
+        Keyword argument:
+        patch_list -- The list of patch file names to replace the script patch list with.
+        """
+        path = os.path.join(self.wpilib_root, f"upstream_utils/{self.name}.py")
+        with open(path, "r") as file:
+            lines = file.readlines()
+
+        for i in range(len(lines) - 1, -1, -1):
```

Comment:
The overall algorithm could use some comments. As far as I can tell, this:

1. Searches backward from the end of the file until it finds `patch_list = `
2. Searches forward until it finds the matching closing square bracket
3. Generates a new Python list of patches
4. Replaces the old Python list